core/jazz_ui/src/app/create-service/create-service/create-service.component.html

```diff
@@ -0,0 +1,211 @@
+<jazz-mob-header (backClick)='closeCreateService()'
+                 class='show-for-mobile'
+                 [icon]="'notification'"
+                 [headText]="'Create a Service'"
+                 [status]="'none'"></jazz-mob-header>
+
+<div class="create-service-menu"
+     [ngClass]="isLoading ? 'body-fix':'body-auto'">
+  <div class="jz-cs-loader" *ngIf="isLoading">
+    <div class="loader">
+      <div class="inner one"></div>
+      <div class="inner two"></div>
+      <div class="inner three"></div>
+    </div>
+  </div>
+  <section class="create-service-header hide-for-mobile">
+    <h4 class="jz-bold-title">
+      Create a Service
+    </h4>
+    <div (click)="closeCreateService()" class="icon-icon-close"></div>
+  </section>
+
+  <ng-container [ngSwitch]="createServiceState">
+    <ng-container *ngSwitchCase="'form'">
+      <create-service-row title="Choose the type of service you want to create" required>
+        <radio-panels [(selected)]="createServiceForm.type" [options]="serviceTypeOptions"></radio-panels>
+      </create-service-row>
+      <create-service-row title="Choose your platform you would like to use" required>
+        <radio-panels [(selected)]="createServiceForm.platform"
+                      [options]="platformOptions"></radio-panels>
+      </create-service-row>
+      <create-service-row title="Choose your runtime"
+                          [type]="createServiceForm.type" serviceTypes="api function" required>
+        <radio-checkbox [(selected)]="createServiceForm.runtime"
+                        [options]="runtimeOptions"></radio-checkbox>
+      </create-service-row>
+      <create-service-row title="Service Name" required
+                          subtitle="Name of your API, Microservice or Website that you want to create.">
+        <div class="jz-relative">
+          <input type="text" class="jz-input"
+                 placeholder="Enter Service Name"
+                 [attr.data-state]="serviceNameMessage"
+                 [(ngModel)]="createServiceForm.serviceName"
+                 (blur)="validateServiceName($event)">
+          <img *ngIf="availabilityLoader" class="jz-input-loader" src="../assets/images/icons/loading.png">
+          <div [ngSwitch]="serviceNameMessage" class="jz-input-message">
```

Comment:
do you think this kind of validation can be in its own component. 
this component itself is too big and is doing a lot. DO you think breaking this into smaller reusable components can make it better?

Reply:
I considered this, but I wanted the flag that determines the message and the message itself to be easy to find and in the same component. I've also already reduced the file by more than half and thought more componenting might be overkill.